### PR TITLE
Add back favorites shortcut

### PR DESCRIPTION
**Pull request by jasonhenriquez**
# Add back favorites shortcut

## Pull Request Type

- [x] Feature Implementation

## Related issue
favorites shortcut removed in #4234
reopens #3536 and #1762

## Description
- Adds back favorites shortcut
  - Fixes broken internal functions (`removeVideos` and `deleteVideoIdsByPlaylistId`) to do so
  - Shortcut does not show on Favorites playlist page (to avoid redundancy)
  - Minor visual improvement: video overlay icons show on `:has(:focus-visible)` instead of `:focus-within`
- Removes Watch Later playlist (as discussed in Matrix developer chat, there is no reason to make Favorites legacy, thus there is no need for a separate "Watch Later" playlist)
- Makes Favorites playlist protected (prevents deletion of playlist, which would be problematic)

## Screenshots 
Screenshot_20240103_175611
Screenshot_20240103_180431
Screenshot_20240103_175507

## Testing 
- Test adding/removing from Favorites using shortcut
- Test removing from Favorites using shortcut removes all duplicates as well
- Test added/removed videos appear properly after restarting FreeTube
- Visual improvement test: Unfavorite a video in Subscriptions using a mouse click, stop hovering over the video, and see the Save to Favorites & Add to Playlist buttons disappear from view. See that they still show when unfavoriting using keyboard navigation & the Space key.

## Desktop

- **OS:** OpenSUSE Tumbleweed
- **OS Version:** 2023xxxx
- **FreeTube version:** 0.19.1

### CODE REVIEW

**Comment by PikachuEXE** [comment]
I can't test this much due to missing playlist
image

That's one of the reasons I allow user to pick the playlist they want to use (no special treatment = most backward compatible)

**Comment by PikachuEXE** [comment]
I am a user who don't need that playlist and the feature (quick bookmark)
As mentioned before I want to be able to disabled the feature and more importantly remove the useless empty playlist
Making it protected is exactly what I don't want

**Comment by jasonhenriquez** [comment]
[See comment under other PR](https://github.com/FreeTubeApp/FreeTube/pull/4518#issuecomment-1876174833) - don't care how it's resolved, but "missing" items need to be addressed. If you don't have a protected Favorites playlist, you need to find some other mechanism for selecting a default bookmarked playlist.

File: src/datastores/handlers/base.js [hunk 1]
```diff
@@ -147,7 +147,7 @@ class Playlists {
   static deleteVideoIdsByPlaylistId(_id, videoIds) {
     return db.playlists.updateAsync(
       { _id },
-      { $pull: { videos: { $in: videoIds } } },
+      { $pull: { videos: { videoId: { $in: videoIds } } } },
```
Comment (PikachuEXE) [inline src/datastores/handlers/base.js:150]: Is this a fix?

Comment (jasonhenriquez) [reply src/datastores/handlers/base.js:150]: Yes, it is. But this function is not in use in your PR, it seems. You can add this fix as well.

**Pull request closed without merging**